### DISCUSSION TITLE
How to propagate compiler from vcpkg toolchain to cmake main build?

### DISCUSSION THREAD

**Discussion by cpp977** [microsoft/vcpkg#28674] [Q&A] [answered]
Hello,

I am using a custom triplet for compiling with clang via an overlay triplet `x64-linux-clang` which sets `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` to the vcpkg x64-linux toolchain file in which a custom command is appended (`set(CMAKE_CXX_COMPILER clang++)`).
This works so far that all packages from vcpkg are compiled with clang. However the main build is still compiled with g++ unless I set additionally `CXX=clang++ cmake ...`.
Is there a way to propagate the compiler information from vcpkg triplet to the main build?

Thanks!

**Comment by Osyotr** [comment, marked as answer]
You need to set `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` in both vcpkg triplet and your cmake cache (e.g. `cmake -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=...`).
